src/ensembl/src/content/app/browser/browser-nav/BrowserNavBarRegionSwitcher.test.tsx

```diff
@@ -16,20 +17,34 @@ jest.mock(
 );
 
 const props = {
+  display: DisplayType.FULL,
+  breakpointWidth: BreakpointWidth.TABLET,
   toggleRegionEditorActive: jest.fn(),
   toggleRegionFieldActive: jest.fn()
 };
 
 describe('BrowserNavBarRegionSwitcher', () => {
+  let wrapper: any;
+
+  beforeEach(() => {
+    wrapper = mount(<BrowserNavBarRegionSwitcher {...props} />);
+  });
+
   afterEach(() => {
     jest.resetAllMocks();
   });
 
-  it('renders correctly', () => {
-    const wrapper = mount(<BrowserNavBarRegionSwitcher {...props} />);
+  describe('rendering', () => {
+    it('renders only region field on smaller screens', () => {
+      expect(wrapper.find(BrowserRegionField).length).toBe(1);
+    });
+
+    it('renders both region field and region editor on big desktop screens', () => {
+      wrapper.setProps({ display: DisplayType.FULL });
 
-    expect(wrapper.find(BrowserRegionEditor).length).toBe(1);
-    expect(wrapper.find(BrowserRegionField).length).toBe(1);
+      expect(wrapper.find(BrowserRegionEditor).length).toBe(1);
+      expect(wrapper.find(BrowserRegionField).length).toBe(1);
```

Comment:
Here and in other places, would it be better to check `expect(wrapper.find(BrowserRegionField)).toHaveLength(1)`?

Reply:
I think either is fine. Both formats are equally readable ("expect components array length to be X" vs "expect components array to have length X"); none is significantly more verbose than the other.